sklearn/neural_network/tests/test_mlp.py

```diff
@@ -633,3 +638,213 @@ def test_n_iter_no_change_inf():
+        assert_allclose(
+            weighted_score,
+            metric(y1, y2, sample_weight=sample_weight * scaling),
+            err_msg="%s sample_weight is not invariant "
+                    "under scaling" % name)
+
+    # Check that if number of samples in y_true and sample_weight are not
+    # equal, meaningful error is raised.
+    error_message = ("Found input variables with inconsistent numbers of "
+                     "samples: [{}, {}, {}]".format(
+                         _num_samples(y1), _num_samples(y2),
+                         _num_samples(sample_weight) * 2))
+    assert_raise_message(ValueError, error_message, metric, y1, y2,
+                         sample_weight=np.hstack([sample_weight,
+                                                  sample_weight]))
+
+
+def test_sample_weight_invariance(n_samples=50):
+    # Test sample weight loss functions directly
+
+    # regression
+
+    # squared loss
+    random_state = check_random_state(0)
+    y_true = random_state.random_sample(size=(n_samples, 1))
+    y_pred = random_state.random_sample(size=(n_samples, 1))
+    check_sample_weight_invariance(name='squared_loss', metric=squared_loss,
+                                   y1=y_true, y2=y_pred)
+
+    # classification
+
+    # binary log loss
+    random_state = check_random_state(0)
+    y_true = random_state.randint(0, 2, size=(n_samples, 1))
+    y_pred = random_state.random_sample(size=(n_samples, 1))
+    check_sample_weight_invariance(name='binary_log_loss',
+                                   metric=binary_log_loss,
+                                   y1=y_true, y2=y_pred)
+
+    # log loss
+    random_state = check_random_state(0)
+    y_true = random_state.randint(0, 5, size=(n_samples, 1))
+    y_pred = random_state.random_sample(size=(n_samples, 1))
+    check_sample_weight_invariance(name='log_loss', metric=log_loss,
+                                   y1=y_true, y2=y_pred)
+
+
+def test_mlp_sample_weight():
+    # validate sample weight behaviour through fit
+    n_samples = 100
+    X = X_digits_binary[:n_samples]
+    y = y_digits_binary[:n_samples]
+    seed = 0
+
+    # validate sample weight and X have the same shape when calling fit
+    # o.w. raise meaningful message
+    sample_weight = np.ones(n_samples-1, dtype=np.float64, order='C')
+    message = ('Shapes of X and sample_weight do not match.')
+    learners = [MLPClassifier(random_state=seed),
+                MLPRegressor(random_state=seed)]
+    for learner in learners:
+        # fit
+        assert_raise_message(ValueError, message, learner.fit, X, y,
+                             sample_weight)
+        # partial fit
+        if isinstance(learner, MLPRegressor):
+            continue
+        assert_raise_message(ValueError, message, learner.partial_fit, X, y,
+                             classes=np.unique(y),
+                             sample_weight=sample_weight)
+
+    # None sample weight or all being 1 predicts same output
```

Comment:
This should already be tested in sklearn/utils/estimator_checks.py in `check_sample_weights_invariance`. You don't need it here

Reply:
Ok, I will explore this module.